firebase-app-distribution/src/test/java/com/google/firebase/app/distribution/FirebaseAppDistributionTest.java

```diff
@@ -415,4 +436,69 @@ public void taskCancelledOnScreenRotation() {
     assertEquals("Update canceled", e.getMessage());
     assertEquals(INSTALLATION_CANCELED, e.getErrorCode());
   }
+
+  @Test
+  public void updateApp_whenCalledMultipleTimesWithAAB_returnsSameUpdateTask() {
+    when(mockSignInStorage.getSignInStatus()).thenReturn(true);
+    AppDistributionReleaseInternal newRelease = TEST_RELEASE_NEWER_AAB_INTERNAL.build();
+    firebaseAppDistribution.setCachedNewRelease(newRelease);
+
+    UpdateTask updateTask1 = firebaseAppDistribution.updateApp(true);
+    UpdateTask updateTask2 = firebaseAppDistribution.updateApp(true);
+
+    assertEquals(updateTask1, updateTask2);
+  }
+
+  @Test
+  public void updateApp_whenCalledMultipleTimesWithApk_returnsSameUpdateTask() {
```

Comment:
I think this test and the won above it might belong in the updateAAB and updateAPK clients, respectively. I don't see a code path in FirebaseAppDistribution where we check for a cached update task

Reply:
Ah since this was in the updateApp tests I moved it over. I didn't see updateApp checking for a cached update task either. Will move the tests to reflect this.